add sorting to Pools

Soot3 opened the pull request:
Adds default sorting logic to the generated pools list.

related to #48104
@jscheffl
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

Soot3 committed:
add sorting to Pools

Adds default sorting logic to the generated pools list.

related to #48104

bbovenzi commented on airflow-core/src/airflow/ui/src/pages/Pools/Pools.tsx:
```diff
@@ -32,8 +33,17 @@ export const Pools = () => {
   const { NAME_PATTERN: NAME_PATTERN_PARAM }: SearchParamsKeysType = SearchParamsKeys;
   const [poolNamePattern, setPoolNamePattern] = useState(searchParams.get(NAME_PATTERN_PARAM) ?? undefined);
+  
+  const { tableURLState } = useTableURLState();
```
Thanks for picking this up.

We need `setTableURLState` from here too and pass it to the table component like so: `<DataTable onStateChange={setTableURLState}`.

Soot3 replied:
@bbovenzi Oh i saw that. would the rest of the datatable structure be needed as well and integrated to the poolbar mapping that is available now.  for example [here](https://github.com/apache/airflow/blob/87c55b51457bf9dafbcbf541ff51940f0455fd15/airflow/ui/src/pages/Variables/Variables.tsx) 
`<DataTable
          columns={columns}
          data={data ? data.variables : []}
          errorMessage={<ErrorAlert error={error} />}
          initialState={tableURLState}
          isFetching={isFetching}
          isLoading={isLoading}
          modelName="Variable"
          onStateChange={setTableURLState}
          total={data ? data.total_entries : 0}
        />
      </Box>`

bbovenzi replied:
Yes, that sounds like a good idea.

Soot3 replied:
Added @bbovenzi. Please let me know if more is needed. Thank you for your time

Soot3 committed:
added more of datatable structure

added default sorting for pools and more of the datatable structure as in variables and connections

bbovenzi commented on airflow-core/src/airflow/ui/src/pages/Pools/Pools.tsx:
```diff
@@ -60,11 +71,16 @@ export const Pools = () => {
+            <Skeleton height="100px" />
+          ) : (
+            data?.pools.map((pool) => <PoolBar key={pool.name} pool={pool} />)
+          )} 
```
```suggestion
          isLoading={isLoading}
```

We already pass the data below

bbovenzi commented on airflow-core/src/airflow/ui/src/pages/Pools/Pools.tsx:
```diff
@@ -60,11 +71,16 @@ export const Pools = () => {
-        ) : (
-          data?.pools.map((pool) => <PoolBar key={pool.name} pool={pool} />)
-        )}
+        <DataTable
```
Since we don't render this as a regular table but more like a list of cards. I would follow what we do in for [AssetEvents](https://github.com/apache/airflow/blob/main/airflow-core/src/airflow/ui/src/components/Assets/AssetEvents.tsx). 

We create a `cardDef` that accepts a `PoolResponse` and renders `<PoolBar />` or `<Skeleton />`. We then pass that cardDef to the DataTable here.

Soot3 replied:
okay. made the requested changes using the asset events structure.

bbovenzi reviewed (changes requested):
Looking better still a few pieces missing. 

Also, there is no button to search. Check out [AssetEvents](https://github.com/apache/airflow/blob/main/airflow-core/src/airflow/ui/src/components/Assets/AssetEvents.tsx) again and we can add a Select component in place of the `<Spacer />` on line 70.

Soot3 committed:
Update airflow-core/src/airflow/ui/src/pages/Pools/Pools.tsx

Co-authored-by: Brent Bovenzi <brent.bovenzi@gmail.com>

Soot3 committed:
added carddef and select component

Soot3 committed:
lint and formatting

Soot3 committed:
another round of lint

Soot3 committed:
linting and formating

Most of the issues involved type definitions in the new function

bbovenzi reviewed (approved):
nice work!
